components/GiftExchangeHeader/GiftExchangeHeader.test.tsx

```diff
@@ -52,3 +130,110 @@ describe('GiftExchangeHeader', () => {
     ).toBeInTheDocument();
   });
 });
+
+describe('GiftExchangeHeader owner permissions', () => {
+
+  beforeEach(() => {
+    // Reset to default owner mock before each test
+    mockGetUser.mockResolvedValue({
+      data: {
+        user: {
+          id: '111', // Default owner ID matching the gift exchange
+        },
+      },
+      error: null,
+    });
+  });
+
+  afterEach(() => {
+    cleanup(); // Clean up after each test
+  });
+
+  it('matching user_id and owner_id should render the draw gift exchange button', async () => {
```

Comment:
These tests should read `It should, it will, it shouldn't, etc.`
```suggestion
  it('should match user_id and owner_id should render the draw gift exchange button', async () => {
```